src/Admin/Pool.php

```diff
@@ -291,6 +292,53 @@ public function getAdminByAdminCode($adminCode)
         return $admin;
     }
 
+    /**
+     * See if an admin with a certain admin code exists.
+     *
+     * @param string $adminCode
+     *
+     * @return bool
+     */
+    public function hasAdminByAdminCode($adminCode)
```

Comment:
final please

Reply:
also typehints

Reply:
Typehints as in `hasAdminByAdminCode(string $adminCode): bool`?

I added this as a discussion point in my PR comment:

> Should passing a non-string argument to Pool::hasAdminByAdminCode() cause a TypeError already in 3.x? Or keep it consistent with Pool::hasAdminByAdminCode() and allow this but trigger a deprecation error?

The latter would mean no typehint, because that would already trigger a TypeError.

If I were to add the typehint, I should also not add:

https://github.com/sonata-project/SonataAdminBundle/blob/1fec5ea4539763ad47323ca589a5067068470944/src/Admin/Pool.php#L304-L313

Reply:
There is no sense for me in provide support for a scenarios that we don't want in a new feature.
Even, shouldn't this method be just a shortcut to call for `getAdminByAdminCode()` wrapped with the corresponding `try/catch`?

Reply:
Ok, was looking for opinions on this. Thanks, changed it.

> Even, shouldn't this method be just a shortcut to call for `getAdminByAdminCode()` wrapped with the corresponding `try/catch`?

Good point :sweat_smile: Also changed that